src/focus.service.ts

```diff
@@ -245,15 +243,15 @@ export class FocusService {
    */
   public scrollSpeed: number | null = 1000;
   // Focus root, the service operates below here.
-  private root: HTMLElement;
+  private root: HTMLElement | null = null;
   public focusRoot: HTMLElement = defaultFocusRoot;
   // The previous rectange that the user had selected.
   private historyRect = defaultRect;
   // Subscription to focus update events.
-  private registrySubscription: Subscription;
+  private registrySubscription?: Subscription;
```

Comment:
What's the difference between using `| null` and an optional parameter, and why the preference for one over the other?

Reply:
? implied ` | undefined`, I generally prefer it because it's shorter.

We currently have a little bit of a mixed bag of `null | undefined | void`, not a big problem at the moment tho.